lib/location/method_gnss.c

```diff
@@ -87,7 +87,7 @@ static struct gps_pgps_request pgps_request;
 static bool running;
 static struct location_gnss_config gnss_config;
 static K_SEM_DEFINE(entered_psm_mode, 0, 1);
-static K_SEM_DEFINE(entered_rrc_idle, 0, 1);
+static K_SEM_DEFINE(entered_rrc_idle, 1, 1);
```

Comment:
This is the part that fixes GNSS not started issue in very bad LTE coverage. All the other changes are related to priority mode.